query-helpers/acs.js

```diff
@@ -0,0 +1,156 @@
+      GROUP BY base
+  )
+
+  /*
+   * An aggregation of enriched selection, joined with base. For true aggregate data selections,
+   * e, m, c, p, and z (sum, m, cv, percent, percent_m, respectively) are recalculated.
+   * For non-aggrenate data selections, the original values are selected using MAX as a noop aggregate function.
+   * is_reliable is calculated for all data selections.
+   * Note: m is coalesced to 0 if the value does not exist in the data
+   * TODO make this fix in the data? make m NOT NULL DEFAULT 0?
+   * Columns: id, sum, m, cv, variable, variablename, base, category, profile, percent, percent_m, is_reliable
+   */
+  SELECT variables.*,
+    --- percent (do not recalculate for non-aggregate selections)---
+    CASE
+      WHEN NOT ${isAggregate(ids)} THEN percent / 100
+      WHEN base_sum = 0 THEN 0
+      WHEN base_sum IS NULL THEN 0
+      ELSE sum / base_sum
+    END AS percent,
+    --- percent_m ---
+    CASE
+      WHEN NOT ${isAggregate(ids)} THEN percent_m / 100
+      WHEN base_sum = 0 THEN 0
+      WHEN base_sum IS NULL THEN 0
+      WHEN POWER(m, 2) - POWER(sum / base_sum, 2) * POWER(base_m, 2) < 0
+        THEN (1 / base_sum) * SQRT(POWER(m, 2) + POWER(sum / base_sum, 2) * POWER(base_m, 2))
+      ELSE (1 / base_sum) * SQRT(POWER(m, 2) - POWER(sum / base_sum, 2) * POWER(base_m, 2))
+    END AS percent_m,
+    --- is_reliable ---
+    CASE
+      WHEN cv < 20 THEN true
+      ELSE false
+    END AS is_reliable
+  FROM (
+    SELECT
+      --- id ---
+      ENCODE(CONVERT_TO(variable, 'UTF-8'), 'base64') AS id,
+      --- sum ---
+      SUM(e) AS sum,
+      --- m (do not recalculate for non-aggregate selections)---
+      --- coalesce null m to 0 ---
+      CASE
+        WHEN SUM(m) IS NULL THEN 0
+        WHEN NOT ${isAggregate(ids)} THEN MAX(m)
+        ELSE SQRT(SUM(POWER(m, 2)))
+      END AS m,
+      --- cv (do not recalculate for non-aggregate selections)---
+      CASE
+        WHEN NOT ${isAggregate(ids)} THEN MAX(c)
+        ELSE (((SQRT(SUM(POWER(m, 2))) / ${CV_CONST}) / NULLIF(SUM(e), 0)) * 100)
+      END AS cv,
+      --- percent (use MAX() as noop agg for non-agg selections; will be recalculated for aggregate selections)--
+      --- coalesce null percent to 0 ---
+      COALESCE(MAX(p), 0) AS percent,
+      --- percent_m (use MAX() as noop agg for non-agg selections; will be recalculated for aggregate selections)---
+      --- coalesce null percent_m to 0 ---
+      COALESCE(MAX(z), 0) AS percent_m,
+      --- variable ---
+      REGEXP_REPLACE(
+        LOWER(variable),
+        '[^A-Za-z0-9]', '_', 'g'
+      ) AS variable,
+      --- variablename ---
+      variablename,
+      --- base ---
+      base,
+      --- category ---
+      REGEXP_REPLACE(
+        LOWER(category),
+        '[^A-Za-z0-9]', '_', 'g'
+      ) AS category,
+      --- domain ---
+      REGEXP_REPLACE(
+        LOWER(domain),
+        '[^A-Za-z0-9]', '_', 'g'
+      ) AS domain,
```

Comment:
Did this not work before? I noticed you changed this from `profile` to `domain`, I'm guessing to match the `domain` column in the ACS tables. Did you change column schema that or was that just something we overlooked?

Reply:
Just looked at the migration files, looks like it was `domain` this whole time. Do you know if this `domain` property is used in our API code anywhere?

Reply:
I don't see it being used anywhere in the front end, I just changed this to keep things consistent with the schema.

Reply:
Gotcha